feat: Add active discovery to subnets

skyecas opened the pull request:
## Description of changes

Introduces the `managed` and `active_discovery` fields to subnets, where previously they were `true` and `undefined`.

Both default to true, and can be specified false with:

```terraform
resource "maas_subnet" "test" {
	active_discovery = false
	cidr		 = 10.10.0.0/24
	managed		 = false
	name		 = "test subnet"
}
```

<!-- A clear and concise description of your changes here. -->

## Issue or ticket link (if applicable)

Resolves: #283 

<!-- 
Please link the issue this PR addresses using GitHub’s keywords to automatically close related issues. See https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

e.g. Resolves: #XXX -->

## Checklist

- [x] I have written a PR title that follows the advice in DEVELOPMENT.md with the title format `type: title`
- [x] I have written a description of the changes in the PR that is clear and concise.
- [x] I have updated the documentation to reflect the changes.
- [x] I have added or updated the tests to reflect the changes.
- [x] I have run the unit tests and they pass.
- [ ] I have run the acceptance tests and they pass.

ibalutoiu-cbsl committed:
Bump gomaasclient

Use the updated `gomaasclient` that drops `Pod` naming in favor of `VMHost`.

ibalutoiu-cbsl committed:
Update VM host machine network and storage params

Properly implement the network and storage parameters for
the `maas_vm_host_machine` resource.

ibalutoiu-cbsl committed:
Refactor instance allocate & deploy params

ibalutoiu-cbsl committed:
Add 'network_interfaces' param to 'maas_instance'

ibalutoiu-cbsl committed:
Add defaults for 'min_cpu_count' and 'min_memory'

ibalutoiu-cbsl committed:
Implement importers for the managed resources

ibalutoiu-cbsl committed:
Merge pull request #1 from ionutbalutoiu/development

Updates & new features

ibalutoiu-cbsl committed:
Add maas_block_device managed resource

ibalutoiu-cbsl committed:
Merge pull request #2 from ionutbalutoiu/development

Add maas_block_device managed resource

ibalutoiu-cbsl committed:
Set TF state only for computed fields

However, make sure those fields are properly set when importing
an existing resource.

ibalutoiu-cbsl committed:
Implement network managed resources

The following network managed resources are added:
  * maas_fabric
  * maas_vlan
  * maas_subnet
  * maas_space

ibalutoiu-cbsl committed:
Implement ip_ranges config to maas_subnet resource

ibalutoiu-cbsl committed:
Implement maas_subnet_ip_range managed resource

ibalutoiu-cbsl committed:
Re-organize examples

Also:
  * Add `maas_subnet_ip_range` managed resources examples
  * Add two more managed machines as examples

ibalutoiu-cbsl committed:
General code cleanup

ibalutoiu-cbsl committed:
Add maas_dns_domain managed resource

ibalutoiu-cbsl committed:
Add maas_dns_record managed resource

ibalutoiu-cbsl committed:
Bump gomaasclient

ibalutoiu-cbsl committed:
Update examples

Add examples for maas_domain and maas_dns_record managed resources

ibalutoiu-cbsl committed:
Update docs

ibalutoiu-cbsl committed:
Add validation for maas_machine power_type

ibalutoiu-cbsl committed:
Remove managed attr from subnet

ibalutoiu-cbsl committed:
Add maas_user managed resource

ibalutoiu-cbsl committed:
Merge pull request #3 from ionutbalutoiu/development

More managed resources, code improvements, and docs updated

ibalutoiu-cbsl committed:
Bump VERSION in Makefile

Commit:
bump gomaasclient version

cpg1111 committed:
Merge pull request #14 from cpg1111/update_go_client

bump gomaasclient version

pvegh-oc committed:
Updating go version to 1.19 updating all dependencies to latest

pvegh-oc committed:
Adding dependabot configuration

pvegh-oc committed:
Merge pull request #17 from maas/updating-go-dependencies

Updated go dependencies.
Updated MAAS client version.
Added dependabot configuration.

trentrosenbaum committed:
Add a test and release workflow to the provider

trentrosenbaum committed:
Update .github/workflows/test.yml

Co-authored-by: Greg Nuttall <102314841+greg-oc@users.noreply.github.com>

Commit:
Merge pull request #20 from maas/introdcuce-github-workflows

Introduces GitHub workflows for test and release

Commit:
Update makefile targets (#19)

* Introduces Makefile targets to clean and allow tool execution over Terraform provider

* Updates Makefile to generate a dev overrides file for local development

pvegh-oc committed:
Adds goreleaser config to support release workflow (#21)

greg-oc committed:
Refactor documentation to use terraform-plugin-docs tool (#18)

* add descriptions to maas_subnet and data source examples

* add examples to new structure for tfplugindocs

* Add attribute descriptions to schema

* add import examples

* adds descriptions for resources

* update maas instance description

* change name in makefile

* WIP add index.md.tmpl

* add CHANGELOG

* remove old examples directory

* runs document generation tool

* change source in provider example

* Updates Makefile to allow docs to be generated and validated

Co-authored-by: Trent Rosenbaum <trent.j.rosenbaum@gmail.com>

trentrosenbaum committed:
Restricts the test workflow to the main branch

pvegh-oc committed:
Documenting the release process

pvegh-oc committed:
Fixing main-master branch naming

pvegh-oc committed:
Fixing typo, rephrasing bulletpoint 1

pvegh-oc committed:
Updating git commands

pvegh-oc committed:
fixing typo

Co-authored-by: Adam Collard <sparkiegeek@users.noreply.github.com>

pvegh-oc committed:
Merge pull request #23 from maas/adding-release-process-documentation

Adding release process documentation

greg-oc committed:
refactor api client references to use maas repo name

greg-oc committed:
add .vscode and vendor directories to .gitignore

greg-oc committed:
adds draft documentation

greg-oc committed:
run tfplugindocs tool

greg-oc committed:
adds small improvements to release guide

greg-oc committed:
rename resource on page_title

greg-oc committed:
Merge branch 'develop' of https://github.com/maas/terraform-provider-maas into draft-documentation

greg-oc committed:
Merge pull request #25 from maas/rename-imports

Refactor API client references to use maas repo name

greg-oc committed:
Merge pull request #27 from maas/release-guide-changes

Slight improvements to the release process guide

greg-oc committed:
updates index.md

greg-oc committed:
remove unnecessary line

greg-oc committed:
Merge pull request #26 from maas/draft-documentation

Adds documentation

greg-oc committed:
Updates CHANGELOG

pvegh-oc committed:
Downgrading go version to 1.18 as requested

greg-oc committed:
adds missing empty last lines

greg-oc committed:
add missing empty lines at file end

greg-oc committed:
adds final missing lines

greg-oc committed:
Update Go version in CHANGELOG

Change Go version to 1.18 in CHANGELOG

greg-oc committed:
Merge pull request #29 from maas/develop

* Add release process guide 
* Add GitHub Actions workflow for releasing
* Add Dependabot configuration
* Update documentation and documentation structure to use terraform-plugin-docs tool
* Refactor API client references to use the "maas" repository
* Update dependencies
* Update Go version to 1.18

Commit:
timeouts for instance and machine

Commit:
Update CHANGELOG.md

Commit:
add e2e test check

dependabot[bot] committed:
Bump goreleaser/goreleaser-action from 3 to 4

Bumps [goreleaser/goreleaser-action](https://github.com/goreleaser/goreleaser-action) from 3 to 4.
- [Release notes](https://github.com/goreleaser/goreleaser-action/releases)
- [Commits](https://github.com/goreleaser/goreleaser-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: goreleaser/goreleaser-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
fix typo

cpg1111 committed:
Merge pull request #41 from cpg1111/add_github_action_for_e2e_tests

add e2e test check

Commit:
bump gomaasclient version for TLS fix

cpg1111 committed:
Merge pull request #47 from cpg1111/bump_gomaasclient_version

bump gomaasclient version for TLS fix

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-docs from 0.13.0 to 0.14.1

Bumps [github.com/hashicorp/terraform-plugin-docs](https://github.com/hashicorp/terraform-plugin-docs) from 0.13.0 to 0.14.1.
- [Release notes](https://github.com/hashicorp/terraform-plugin-docs/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-docs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-docs/compare/v0.13.0...v0.14.1)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-docs
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

sparkiegeek committed:
Merge pull request #43 from maas/dependabot/github_actions/goreleaser/goreleaser-action-4

Bump goreleaser/goreleaser-action from 3 to 4

Commit:
Merge pull request #34 from ageoffron/add_timeouts_support

Add timeouts support

dependabot[bot] committed:
Bump actions/setup-go from 3 to 4

Bumps [actions/setup-go](https://github.com/actions/setup-go) from 3 to 4.
- [Release notes](https://github.com/actions/setup-go/releases)
- [Commits](https://github.com/actions/setup-go/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/setup-go
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mastier committed:
feat: add enable_hw_sync to deploy params (#59)

* add enable_hw_sync to deploy params
* bump github.com/maas/gomaasclient

---------

Co-authored-by: Anton Troyanov <anton@troyanov.net>

troyanov committed:
Release v1.2.0

troyanov committed:
ci: test if docs are up to date

troyanov committed:
docs: generate & update docs

troyanov committed:
docs: fix min go version in README

go.mod has go 1.18

dependabot[bot] committed:
Bump github.com/stretchr/testify from 1.8.0 to 1.8.2

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.8.0 to 1.8.2.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.8.0...v1.8.2)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/stretchr/testify from 1.8.2 to 1.8.3

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.8.2 to 1.8.3.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.8.2...v1.8.3)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/stretchr/testify from 1.8.3 to 1.8.4

Bumps [github.com/stretchr/testify](https://github.com/stretchr/testify) from 1.8.3 to 1.8.4.
- [Release notes](https://github.com/stretchr/testify/releases)
- [Commits](https://github.com/stretchr/testify/compare/v1.8.3...v1.8.4)

---
updated-dependencies:
- dependency-name: github.com/stretchr/testify
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.21.0 to 2.27.0

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.21.0 to 2.27.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.21.0...v2.27.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
remove e2e check

skyecas committed:
Merge pull request #73 from maas/dependabot/go_modules/github.com/hashicorp/terraform-plugin-sdk/v2-2.27.0

Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.21.0 to 2.27.0

skyecas committed:
Update go.mod

Bump to go 1.19

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-docs from 0.14.1 to 0.16.0

Bumps [github.com/hashicorp/terraform-plugin-docs](https://github.com/hashicorp/terraform-plugin-docs) from 0.14.1 to 0.16.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-docs/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-docs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-docs/compare/v0.14.1...v0.16.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-docs
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

skyecas committed:
force update docs

skyecas committed:
Merge pull request #74 from maas/dependabot/go_modules/github.com/hashicorp/terraform-plugin-docs-0.16.0

Bump github.com/hashicorp/terraform-plugin-docs from 0.14.1 to 0.16.0

skyecas committed:
Merge pull request #78 from maas/bump-gomod-min-version

Update go.mod

r00ta committed:
bump gomaasclient/gomaasapi to latest version to include retry improvements

r00ta committed:
trigger pipeline

r00ta committed:
trigger pipeline

tmerten committed:
Merge pull request #85 from r00ta/bump-gomaasclient

bump gomaasclient/gomaasapi to latest version to include retry improvements

skyecas committed:
Merge pull request #76 from cpg1111/remove_e2e_check_in_favor_of_new_reporting

remove e2e check

skatsaounis committed:
Add missing tag fields (#90)

skatsaounis committed:
Bump gomaasclient/gomaasapi to latest version (#92)

troyanov committed:
all: update Go module to Go 1.20 (#87)

In order to use recent version of Terraform plugin SDK [0] we need to
bump our minimal Go version to 1.20 because of this PR [1] being merged.

Also each major Go release is supported until there are two newer major releases [2].

[0]: github.com/hashicorp/terraform-plugin-sdk/v2
[1]: https://github.com/maas/terraform-provider-maas/pull/83
[2]: https://go.dev/doc/devel/release#policy

dependabot[bot] committed:
Bump goreleaser/goreleaser-action from 4 to 5 (#86)

Bumps [goreleaser/goreleaser-action](https://github.com/goreleaser/goreleaser-action) from 4 to 5.
- [Release notes](https://github.com/goreleaser/goreleaser-action/releases)
- [Commits](https://github.com/goreleaser/goreleaser-action/compare/v4...v5)

---
updated-dependencies:
- dependency-name: goreleaser/goreleaser-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
Bump crazy-max/ghaction-import-gpg from 5 to 6 (#84)

Bumps [crazy-max/ghaction-import-gpg](https://github.com/crazy-max/ghaction-import-gpg) from 5 to 6.
- [Release notes](https://github.com/crazy-max/ghaction-import-gpg/releases)
- [Commits](https://github.com/crazy-max/ghaction-import-gpg/compare/v5...v6)

---
updated-dependencies:
- dependency-name: crazy-max/ghaction-import-gpg
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/checkout from 3 to 4 (#82)

Bumps [actions/checkout](https://github.com/actions/checkout) from 3 to 4.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.27.0 to 2.29.0 (#83)

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.27.0 to 2.29.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.27.0...v2.29.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
Release v1.3.0

skatsaounis committed:
Fix parsing of power parameters (#96)

troyanov committed:
ci: add lint job with gofmt (#99)

troyanov committed:
fix: tfproviderlint/x targets (#100)

troyanov committed:
ci: fix gofmt param, so it scans all dirs (#103)

r00ta committed:
bump gomaasclient to 20231012

r00ta committed:
Merge pull request #104 from r00ta/bump_gomaasclient_20231012

Bump gomaasclient to 20231012

troyanov committed:
chore: update changelog with release v1.3.1 (#107)

troyanov committed:
ci: run terraform linters (#102)

* chore(deps): bump bflad/tfproviderlint to v0.29.0

* chore(lint): R014

* chore(lint): XS001

* chore(lint): XS002

* chore(lint): S018

* chore(lint): R001

* ci: run terraform linters

troyanov committed:
all: drop support of Terraform EOL versions (#108)

1.4 is also EOL, however it is still widely used.

https://support.hashicorp.com/hc/en-us/articles/360021185113

troyanov committed:
ci: special labels to trigger integration tests

Introduce label based workflow.
MAAS maintainers have to apply special label in order to trigger
integration tests using real MAAS installation.

Labels are automatically removed when new commits are pushed to the PR
branch.

skatsaounis committed:
Add support for TLS configuration options (#101)

dependabot[bot] committed:
Bump hashicorp/setup-terraform from 2 to 3

Bumps [hashicorp/setup-terraform](https://github.com/hashicorp/setup-terraform) from 2 to 3.
- [Release notes](https://github.com/hashicorp/setup-terraform/releases)
- [Changelog](https://github.com/hashicorp/setup-terraform/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/setup-terraform/compare/v2...v3)

---
updated-dependencies:
- dependency-name: hashicorp/setup-terraform
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/github-script from 6 to 7

Bumps [actions/github-script](https://github.com/actions/github-script) from 6 to 7.
- [Release notes](https://github.com/actions/github-script/releases)
- [Commits](https://github.com/actions/github-script/compare/v6...v7)

---
updated-dependencies:
- dependency-name: actions/github-script
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

skatsaounis committed:
Fix maas_instance updates (#117)

ianmcmahon committed:
Add datasources for machine and physical nic (#7)


---------

Co-authored-by: Ian McMahon <facetious.ian@gmail.com>
Co-authored-by: Anton Troyanov <anton@troyanov.net>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.29.0 to 2.30.0 (#112)

Bumps [github.com/hashicorp/terraform-plugin-sdk/v2](https://github.com/hashicorp/terraform-plugin-sdk) from 2.29.0 to 2.30.0.
- [Release notes](https://github.com/hashicorp/terraform-plugin-sdk/releases)
- [Changelog](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/terraform-plugin-sdk/compare/v2.29.0...v2.30.0)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/terraform-plugin-sdk/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
Deploy a machine based on its system_id (#118)

ironashram committed:
chore: replace deprecated package (#114)

skatsaounis committed:
Consume latest gomaasclient and fix drifts (#121)

skatsaounis committed:
Add device datasource (#119)

dependabot[bot] committed:
Bump actions/setup-go from 4 to 5 (#123)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/maas/gomaasclient from 0.0.0-20231117143343-dfb55c0a4c5e to 0.1.0 (#124)

Bumps
[github.com/maas/gomaasclient](https://github.com/maas/gomaasclient)
from 0.0.0-20231117143343-dfb55c0a4c5e to 0.1.0.

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
Release v2.0.0 (#131)

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.30.0 to 2.31.0 (#133)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
fix: deleting VM hosts of dynamic machines do not try release (#138)

dependabot[bot] committed:
Bump github.com/maas/gomaasclient from 0.1.0 to 0.2.0 (#139)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
fix: use MAAS defaults for VM host provisioning (#142)

skatsaounis committed:
feat: add resource pool resources (#134)

troyanov committed:
ci: autoclose stale issues (#144)

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-docs from 0.16.0 to 0.18.0 (#148)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
Bump github.com/maas/gomaasclient from 0.2.0 to 0.3.0 (#149)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.31.0 to 2.32.0 (#150)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/maas/gomaasclient from 0.3.0 to 0.4.0 (#156)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
docs: improve documentation (#157)

skatsaounis committed:
Release v2.1.0 (#158)

dependabot[bot] committed:
Bump github.com/stretchr/testify from 1.8.4 to 1.9.0 (#160)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skyecas committed:
feat: add configurable create timeout for VM hosts (#164)

dependabot[bot] committed:
Bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.32.0 to 2.33.0 (#159)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

poddm committed:
feat: add interface resources for bonds, bridges, and VLANs (#77)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

skatsaounis committed:
Release v2.2.0 (#168)

skatsaounis committed:
fix: properly import physical interfaces (#169)

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-docs from 0.18.0 to 0.19.0 (#171)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-docs from 0.19.0 to 0.19.2 (#174)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
ci: add Canonical CLA check (#182)

skatsaounis committed:
fix: manage existing physical interfaces from commissioning (#178)

dependabot[bot] committed:
chore(deps): bump github.com/bflad/tfproviderlint from 0.29.0 to 0.30.0 (#179)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-docs from 0.19.2 to 0.19.3 (#183)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
bug: respect given timeouts for create/delete operations (#185)

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-docs from 0.19.3 to 0.19.4 (#186)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.33.0 to 2.34.0 (#180)

dependabot[bot] committed:
chore(deps): bump github.com/juju/gomaasapi/v2 from 2.2.0 to 2.3.0 (#188)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump goreleaser/goreleaser-action from 5 to 6 (#187)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github.com/maas/gomaasclient from 0.4.0 to 0.5.0 (#184)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

troyanov committed:
chore(deps): replace/bump github.com/maas/gomaasclient (#194)

skatsaounis committed:
Release v2.3.0 (#207)

skatsaounis committed:
ci: fix goreleaser (#208)

troyanov committed:
ci: fix goreleaser config (#209)

troyanov committed:
chore: proof of ownership for TF registry redirect (#212)

troyanov committed:
chore: remove .terraform-registry (#214)

troyanov committed:
ci: use `pull_request_target` for cla-check (#219)

troyanov committed:
ci: use stable Go compiler version for releases (#211)

skatsaounis committed:
fix: correctly parse vlan field of VLAN interfaces (#227)

alexsander-souza committed:
chore: bump minimum Go version in go.mod to 1.21 (#231)

skatsaounis committed:
docs: bump minimum Go version in README.md to 1.21 (#232)

skatsaounis committed:
fix: tag should query MAAS machines once (#228)

skatsaounis committed:
chore(deps): bump github.com/canonical/gomaasclient from 0.6.0 to 0.7.0 (#233)

skatsaounis committed:
feat: new datasource for maas_rack_controller (#229)

WhimsyWolfe committed:
feat: ephemeral deployment (#218)

skatsaounis committed:
Release v2.4.0 (#235)

dependabot[bot] committed:
chore(deps): bump github.com/stretchr/testify from 1.9.0 to 1.10.0 (#245)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/crypto from 0.23.0 to 0.31.0 (#248)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.25.0 to 0.33.0 (#250)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
chore: update Go module to Go 1.22 (#251)

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-docs from 0.19.4 to 0.20.1 (#246)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.34.0 to 2.35.0 (#242)

skyecas committed:
feat: add Boot source Data source (#253)

Co-authored-by: Stamatis Katsaounis <katsaouniss@gmail.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.35.0 to 2.36.0 (#256)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skyecas committed:
refactor: Reduce testing interface name length (#262)

skyecas committed:
feat: add maas_boot_source resource (#255)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
chore(deps): bump canonical/has-signed-canonical-cla from 1 to 2 (#266)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github.com/bflad/tfproviderlint from 0.30.0 to 0.31.0 (#265)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
fix: properly unlink existing links before creating a link (#268)

tobiasdemendonca committed:
docs: create DEVELOPMENT.md and PR template (#269)

skatsaounis committed:
fix: use unique MAC addresses for tf acc interfaces (#271)

skyecas committed:
feat: Add boot source selection (#264)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

peterctl committed:
feat: add resource and data source for MAAS zones (#258)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.36.0 to 2.36.1 (#270)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tobiasdemendonca committed:
fix: IP ranges cannot detect changes (#272)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

skatsaounis committed:
fix: determine the correct partition order (#263)

dependabot[bot] committed:
chore(deps): bump github.com/canonical/gomaasclient from 0.7.0 to 0.8.0 (#277)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-docs from 0.20.1 to 0.21.0 (#276)

tobiasdemendonca committed:
feat(vm_host): add deploy params to maas_vm_host resource (#275)

Co-authored-by: Stamatis Katsaounis <katsaouniss@gmail.com>

tobiasdemendonca committed:
fix: 195 dns record removal (#280)

Fixes the issue of accidental deletion of both DNS resources that share a common IP address when one is deleted. This also fixes an error caused by a race condition when destroying both resources.

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/go-cty from 1.4.1-0.20200414143053-d3edf31b6320 to 1.4.1 (#282)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tobiasdemendonca committed:
feat: interfaces device support (#284)

Co-authored-by: Bertrand Lanson <bertrand.lanson@protonmail.com>
Co-authored-by: Bertrand Lanson <89710239+ednxzu@users.noreply.github.com>

- The resource `maas_network_interface_link` now works with machines and devices, using the new `device` key. 
- Devices cannot use the resource `maas_network_interface_physical`, as network interfaces should be configured on the device resource itself. This aligns with real world scenarios, that network interfaces are part of the physical deployment profile of the device. 
- Device `Update` now updates network interfaces if their mac addresses change, as this is the only non-computed property.

tobiasdemendonca committed:
feat: 230 extend tagging to support nics (#290)

An additional resource to manage the tags of network interfaces. This resource must not be used in conjunction with the nested tag keys in resources such as `maas_network_interface_physical`. Other examples of additional supporting resources that follow this design pattern can be found in the [AWS provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/resource-tagging#ignoring-changes-to-specific-tags).

tobiasdemendonca committed:
feat: 230-extend-tagging-to-support-block-devices (#292)

An additional resource to manage the tags of block devices. This resource must not be used in conjunction with the nested tag keys in resources such as `maas_block_devices`. See https://github.com/canonical/terraform-provider-maas/pull/290 for previous work on this issue.

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.35.0 to 0.36.0 (#285)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

troyanov committed:
ci: autoclose uses maas-github-workflows (#295)

tobiasdemendonca committed:
fix: avoid subnet cidr collisions (#296)

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/go-cty from 1.4.1 to 1.5.0 (#289)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skatsaounis committed:
fix: check that boot_source url is one of the expected (#300)

skatsaounis committed:
fix: maas_boot_source proper delete (#301)

skatsaounis committed:
chore: add golangci-lint (#287)

tobiasdemendonca committed:
chore(deps): bump github.com/canonical/gomaasclient from 0.8.0 to 0.9.0 (#302)

skyecas committed:
feat: Introduce volume group resource (#293)

skatsaounis committed:
feat: add maas_vlan_dhcp resource (#247)

Co-authored-by: Toby de Mendonça <toby.demendonca@canonical.com>

Implement a resource to configure DHCP on a VLAN. This replaces the need to use the `dhcp_on` parameter in the `maas_vlan` resource, which does not appropriately capture all subnet/iprange resource dependencies, and lacks the ability to set primary/secondary racks or relay vlans. This new resource allows this. See `examples/resources/maas_vlan_dhcp` for intended use.

> [!NOTE] 
> the `dchp_on` parameter of the `maas_vlan` resource will be removed in a future PR. This is because consideration needs to be taken to ensure upgrading the provider with an existing state file works without error.

These tests will likely fail once https://bugs.launchpad.net/maas/+bug/2107168 is fixed

ricardbejarano committed:
feat: add `maas_machines` data source (#299)

Signed-off-by: Ricard Bejarano <ricard@bejarano.io>

tobiasdemendonca committed:
chore: add PR template instructions (#308)

ricardbejarano committed:
feat: add `maas_devices` data source (#298)

Signed-off-by: Ricard Bejarano <ricard@bejarano.io>

skyecas committed:
feat: add datasource for MAAS Boot Resources (#306)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

tobiasdemendonca committed:
feat: 189-storage-device-details-of-enlisted-machines (#307)

dependabot[bot] committed:
chore(deps): bump golang.org/x/net from 0.36.0 to 0.38.0 (#309)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Toby de Mendonça <toby.demendonca@canonical.com>

skyecas committed:
chore(deps): bump github.com/canonical/gomaasclient from 0.9.0 to 0.10.0 (#311)

tobiasdemendonca committed:
feat: 72-erase-disks-when-destroying-instances (#310)

Add ability to erase disks when destroying the resource `maas_instance`.

skyecas committed:
feat: Introduce LVM resource management (#294)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

skyecas committed:
feat: Add boot resource operations (#274)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

tobiasdemendonca committed:
chore: automate-release (#312)

dependabot[bot] committed:
chore(deps): bump github.com/canonical/gomaasclient from 0.10.0 to 0.11.0 (#315)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump golangci/golangci-lint-action from 7 to 8 (#316)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

skyecas committed:
add managed and active disocvery params

skyecas committed:
add to the datasource too

skyecas committed:
add basic tests

SK1Y101 commented:
Requires latest version of GoMAASClient to be included

dependabot[bot] committed:
chore(deps): bump github.com/canonical/gomaasclient from 0.11.0 to 0.12.0 (#318)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skyecas committed:
Merge branch 'master' into add-active-discovery-to-subnets

skyecas committed:
feat: Add RAID resources (#313)

skyecas committed:
Merge branch 'master' into add-active-discovery-to-subnets

skyecas committed:
ip range and managed conflict with eachother

tobiasdemendonca committed:
feat: add-missing-vm-host-params (#314)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
chore(deps): bump github.com/hashicorp/terraform-plugin-sdk/v2 from 2.36.1 to 2.37.0 (#321)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

skyecas committed:
Merge branch 'master' into add-active-discovery-to-subnets

tobiasdemendonca commented on maas/resource_maas_subnet_test.go:
```diff
@@ -0,0 +1,166 @@
+) string {
+	return fmt.Sprintf(`
+resource "maas_subnet" "test" {
+	active_discovery = %t
+	allow_dns		 = %t
+	allow_proxy		 = %t
+	cidr			 = %q
+	dns_servers		 = ["8.8.8.8"]
+	gateway_ip		 = %q
+	managed			 = %t
+	name			 = %q
+	fabric			 = "0"
+	vlan			 = "0"
+}`, activeDiscovery, allowDNS, allowProxy, cidr, gatewayIP, managed, name)
```
Nit: turn tabs -> spaces for config indent

skyecas committed:
tab to space

tobiasdemendonca commented on maas/resource_maas_subnet.go:
```diff
@@ -85,9 +93,10 @@ func resourceMAASSubnet() *schema.Resource {
-				Description: "A set of IP ranges configured on the new subnet. Parameters defined below. This argument is processed in [attribute-as-blocks mode](https://www.terraform.io/docs/configuration/attr-as-blocks.html).",
+				Type:          schema.TypeSet,
+				Optional:      true,
+				ConflictsWith: []string{"managed"},
```
I am in the process of checking what the implications of this are but you might be faster!

- Not sure if the changing managed to unmanaged definitely changes this behaviour on the fly, and all ip range meanings invert.
- I assume that having an unmanged subnet on a vlan with our new dhcp resource enabled is fine but again need to check this.

tobiasdemendonca reviewed (changes requested):
I think this does need a change but I'm being slow to test out these scenarios so just wanted to flag this early incase you also have time to think about the edge cases

tobiasdemendonca committed:
feat: add ssh key resource (#325)

Co-authored-by: Stamatis Katsaounis <katsaouniss@gmail.com>

skyecas committed:
refactor: Update user resource (#323)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

skyecas committed:
feat: add Package Repositories Resource (#327)

Co-authored-by: Toby de Mendonça <toby.demendonca@canonical.com>
Co-authored-by: Stamatis Katsaounis <katsaouniss@gmail.com>

tobiasdemendonca committed:
feat: add maas_configuration resource (#326)

Add the maas_configuration resource to control maas settings as key value pairs.

tobiasdemendonca committed:
feat: add settings datasource (#333)

skyecas committed:
feat: Add Package repo datasource (#331)

Co-authored-by: Toby de Mendonça <toby.demendonca@canonical.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

dependabot[bot] committed:
chore(deps): bump github.com/cloudflare/circl from 1.6.0 to 1.6.1 (#332)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

seb54000 committed:
feat: add a datasource for vm_host (#181)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>
Co-authored-by: Toby de Mendonça <toby.demendonca@canonical.com>

dependabot[bot] committed:
chore(deps): bump golang.org/x/crypto from 0.38.0 to 0.39.0 (#335)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump github.com/canonical/gomaasclient from 0.13.0 to 0.15.0 (#337)

tobiasdemendonca committed:
docs: update README.md (#322)

Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

seb54000 committed:
feat: Add static_route mgmt (#305)

Co-authored-by: Jack Lloyd-Walters <jack.lloyd-walters@canonical.com>
Co-authored-by: Stamatis Katsaounis <stamatis.katsaounis@canonical.com>

skatsaounis committed:
feat: maas_node_script resource (#328)

tobiasdemendonca committed:
fix: 338 restricting package repository attributes (#339)

skyecas committed:
Merge branch 'master' into add-active-discovery-to-subnets
